[17.0][MIG] hr_employee_lastnames: Migration to 17.0

andreagidaltig opened the pull request:
- Rename module from hr_employee_lastnames → hr_employee_second_lastname
  in order to be consistent with partner_second_lastname module.
- Fix some new lints.
- Update incoming parameters of post_init_hook method since `cr, registry`
  were replaced by `env` in [1].
- Introduce the use of new onchange method [2] (renamed in [3]) to
  return the fields specification from a view description.

[1] https://github.com/odoo/odoo/commit/b4a7996
[2] https://github.com/odoo/odoo/commit/f5e6494

MiquelRForgeFlow committed:
[10.0][ADD] hr_holidays_settings

MiquelRForgeFlow committed:
[FIX] add company_id in view

MiquelRForgeFlow committed:
[FIX][hr_holidays_settings] Make 'Leave Types' view accessible (#408)

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

tarteo committed:
[MIG] hr_holidays_settings: Migration to 11.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update hr_holidays_settings.pot

fshahy committed:
Translated using Weblate (Persian)

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-11.0/hr-11.0-hr_holidays_settings
Translate-URL: https://translation.odoo-community.org/projects/hr-11-0/hr-11-0-hr_holidays_settings/fa/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100,0% (4 of 4 strings)

Translation: hr-11.0/hr-11.0-hr_holidays_settings
Translate-URL: https://translation.odoo-community.org/projects/hr-11-0/hr-11-0-hr_holidays_settings/pt_BR/

anandkansagra committed:
[MIG] hr_holidays_settings: Migration to 12.0.

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update hr_holidays_settings.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-12.0/hr-12.0-hr_holidays_settings
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_settings/

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-12.0/hr-12.0-hr_holidays_settings
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_settings/es/

Commit:
Translated using Weblate (French)

Currently translated at 75.0% (3 of 4 strings)

Translation: hr-12.0/hr-12.0-hr_holidays_settings
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_settings/fr/

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (4 of 4 strings)

Translation: hr-12.0/hr-12.0-hr_holidays_settings
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_settings/de/

alexey-pelykh committed:
[FIX] hr_holidays_settings: duplicate Leave Types menuitem

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update hr_holidays_settings.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
hr_holidays_settings 12.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-12.0/hr-12.0-hr_holidays_settings
Translate-URL: https://translation.odoo-community.org/projects/hr-12-0/hr-12-0-hr_holidays_settings/

Commit:
[ADD] hr_employee_lastnames V10.0

Add module to split the last name in to father's last name and mother's
last name.

This module takes the employee name, splits it into three fields, and
assigns each part of the name according to the `partner_names_order`
parameter when calling the inverse name method

OCA-git-bot committed:
hr_employee_lastnames 10.0.2.0.0

luistorresm committed:
[MIG] hr_employee_lastnames: Migration to 11.0

luistorresm committed:
[MIG] : Migration to 12.0

luistorresm committed:
[MIG] hr_employee_lastnames: Migration to 13.0

luistorresm committed:
[MIG] hr_employee_lastnames: Migration to 14.0

Commit:
[REF] hr_employee_lastnames: Removed partner_second_lastname dependency.

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

OCA-git-bot committed:
hr_employee_lastnames 14.0.2.0.0

alexey-pelykh committed:
[UPD] Brainbean Apps => CorporateHub

OCA-git-bot committed:
[UPD] README.rst

britoederr committed:
Added translation using Weblate (Portuguese (Brazil))

Commit:
Added translation using Weblate (Catalan)

Commit:
Translated using Weblate (Catalan)

Currently translated at 71.4% (5 of 7 strings)

Translation: hr-14.0/hr-14.0-hr_employee_lastnames
Translate-URL: https://translation.odoo-community.org/projects/hr-14-0/hr-14-0-hr_employee_lastnames/ca/

hugho-ad committed:
[FIX] hr_employee_lastnames: Move fields to common employee

Starting from 14.0, employee model was split into three models:
- `hr.employee.public`
- `hr.employee.private`
- `hr.employee` (a common model)

When this module was migrated, such  split was not taken into account.

This commit moves fields to the common model, so they don't fail when
read from the public employee model.

Co-Authored-By: Luis González <lgonzalez@vauxoo.com>

OCA-git-bot committed:
hr_employee_lastnames 14.0.3.0.0

luistorresm committed:
[MIG] hr_employee_lastnames: Migration to 15.0

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

Commit:
[IMP] hr_holidays_settings: black, isort, prettier

Commit:
[MIG] hr_holidays_settings: Migration to 14.0

Commit:
[UPD] Update hr_holidays_settings.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
[MIG] hr_holidays_settings: Migration to 15.0

Commit:
[UPD] Update hr_holidays_settings.pot

OCA-git-bot committed:
[UPD] README.rst

douglascstd committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (8 of 8 strings)

Translation: hr-15.0/hr-15.0-hr_employee_lastnames
Translate-URL: https://translation.odoo-community.org/projects/hr-15-0/hr-15-0-hr_employee_lastnames/pt_BR/

douglascstd committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (3 of 3 strings)

Translation: hr-15.0/hr-15.0-hr_holidays_settings
Translate-URL: https://translation.odoo-community.org/projects/hr-15-0/hr-15-0-hr_holidays_settings/pt_BR/

luistorresm committed:
[MIG] hr_employee_lastnames: Migration to 16.0

Commit:
[UPD] Update hr_employee_lastnames.pot

OCA-git-bot committed:
[UPD] README.rst

FernandoRomera committed:
[16.0][FIX] test refresh->invalidate_recordset

OCA-git-bot committed:
hr_employee_lastnames 16.0.1.0.1

badbole committed:
Added translation using Weblate (Croatian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: hr-15.0/hr-15.0-hr_holidays_settings
Translate-URL: https://translation.odoo-community.org/projects/hr-15-0/hr-15-0-hr_holidays_settings/it/

Commit:
[MIG] hr_holidays_settings: Migration to 16.0

Commit:
[UPD] Update hr_holidays_settings.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-16.0/hr-16.0-hr_holidays_settings
Translate-URL: https://translation.odoo-community.org/projects/hr-16-0/hr-16-0-hr_holidays_settings/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: hr-16.0/hr-16.0-hr_employee_lastnames
Translate-URL: https://translation.odoo-community.org/projects/hr-16-0/hr-16-0-hr_employee_lastnames/es/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-16.0/hr-16.0-hr_employee_lastnames
Translate-URL: https://translation.odoo-community.org/projects/hr-16-0/hr-16-0-hr_employee_lastnames/

Commit:
[UPD] Update hr_employee_firstname.pot

OCA-git-bot committed:
[BOT] post-merge updates

andreagidaltig commented:
Hi @luisg123v could you review this please?

cc. @luistorresm

Commit:
[MIG] hr_holidays_settings: Migration to 17.0

FernandoRomera commented:
@andreagidaltig 
Could we change the name of the module to synchronize it with partner_second_lastname?.

Then "hr_employee_lastnames" -> "hr_employee_second_lastname"

Commit:
[UPD] Update hr_holidays_settings.pot

OCA-git-bot committed:
[BOT] post-merge updates

FrancoMaxime commented on hr_employee_lastnames/models/hr_employee.py:
```diff
@@ -0,0 +1,160 @@
+            else:
+                lastname2 = self.lastname2
+            vals["name"] = self._get_name_lastnames(lastname, firstname, lastname2)
+        elif vals.get("name"):
```
Hi @andreagidaltig, i created a PR fixing some issues in the v14 module ( #1283) I could reproduce the bug on the runboat v17. 
To reproduce you can create an employee with a first name equals to " Name 1 Name 2 name 3" and a  lastname equals to "name4"  if you export the data (including the external_id) and importing again the data, the name will be changed. 

Could you fix it or cherry pick my commit? Thanks

luisg123v replied:
@FrancoMaxime since your PR is not merged yet, it shouldn't be included in the migration (because it's still subject to changes).

FrancoMaxime replied:
@luisg123v Thanks, I hadn’t thought of that!

luisg123v commented:
@FernandoRomera 

> Could we change the name of the module to synchronize it with partner_second_lastname?.

Not sure if that's a good idea. That will "break" systems being migrated from previous versions, because the old module would result uninstallable and the new one not installed.

luisg123v commented on hr_employee_second_lastname/__init__.py:
```diff
@@ -0,0 +1,2 @@
+from . import models
+from .hooks import post_init_hook
```
Could you rename hooks file in the previous version, please?

andreagidaltig committed:
[IMP] hr_employee_lastnames: rename hooks file

Rename file of hooks to follow guidelines.

FernandoRomera commented:
> @FernandoRomera
> 
> > Could we change the name of the module to synchronize it with partner_second_lastname?.
> 
> Not sure if that's a good idea. That will "break" systems being migrated from previous versions, because the old module would result uninstallable and the new one not installed.

@luisg123v 
Don't worry, there is already a procedure for this in OpenUpgrade.

If you want we can propose the PR in OpenUpgrade.



apriory.py in OpenUpgrade

Renamed modules is a mapping from old module name to new module name
renamed_modules = {
    ...
    # OCA/hr
    "hr_employee_lastnames": "hr_employee_second_lastname",
    ...

This PR is a example: https://github.com/OCA/OpenUpgrade/pull/4246/files

luisg123v commented:
Got it. I think module renaming should be on a separate commit, though, before running `pre-commit`

@moylop260 what do you think about renaming the module?

moylop260 commented:
I'm not a fan to rename modules

But I don't have the enough context to do it

Consider that you will need to change all the xmlids and you will need to change all the depends from the manifest of modules using this module

OCA-git-bot committed:
[BOT] post-merge updates

FernandoRomera commented:
> Don't worry, there is already a procedure for this in OpenUpgrade.

@moylop260
Don't worry, there is already a procedure for all changes in OpenUpgrade.



> I'm not a fan to rename modules
> 
> But I don't have the enough context to do it
> 
> Consider that you will need to change all the xmlids and you will need to change all the depends from the manifest of modules using this module

@moylop260
Don't worry, there is already a procedure for all changes in OpenUpgrade.

luisg123v commented:
> Don't worry, there is already a procedure for all changes in OpenUpgrade.

Not for this:

> and you will need to change all the depends from the manifest of modules using this module

:smile:

FernandoRomera commented:
> > Don't worry, there is already a procedure for all changes in OpenUpgrade.
> 
> Not for this:
> 
> > and you will need to change all the depends from the manifest of modules using this module
> 
> 😄

You are right, but now is the time to do it and make the nomenclature of the modules consistent.

in the first line of the commit you indicate that you synchronize with the module partner_second_lastname

Therefore partner_contact is like this:
partner_firstname
partner_second_lastname

and in hr
hr_employee_firstname
hr_employee_second_lastname

This is my last comment for it, thanks for all the work done.

andreagidaltig committed:
[MOV] hr_employee_second_lastname: rename module

Module is renamed from hr_employee_lastnames → hr_employee_second_lastname
in order to be consistent with partner_second_lastname module.

andreagidaltig committed:
[IMP] hr_employee_second_lastname: pre-commit stuff

andreagidaltig committed:
[MIG] hr_employee_second_lastname: Migration to 17.0

- Fix some new lints.
- Update incoming parameters of post_init_hook method since `cr, registry`
  were replaced by `env` in [1].
- Deprecate the use of _onchange_spec method and use the method new method
  _get_fields_spec to return the fields specification from a view
  description since it was introduced in [2].

[1] https://github.com/odoo/odoo/commit/b4a7996
[2] https://github.com/odoo/odoo/commit/f5e6494

andreagidaltig commented:
Hi @luisg123v could you review this, please? I just added the commit to rename the module.

luisg123v commented:
/ocabot migration hr_employee_lastnames

luisg123v commented on hr_employee_second_lastname/__init__.py:
Commit message:
- AFAIK tag for module renaming is `[MOV]`
- Commit message doesn't mention where the module is being renamed from. It should include both from and to, e.g.:
  `Module is renamed from hr_employee_lastnames → hr_employee_second_lastname inorder to be consistent with...`

luisg123v commented on hr_employee_second_lastname/__manifest__.py:
```diff
@@ -1,7 +1,7 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Commit message:
- sicronize -> synchronize. Besides, this would only go in PR description because it's not done on this commit anymore
- "Rename hooks file to follow guidelines.": not done on this commit anymore. This was done on v16, so not part of the migration anymore (hence should be removed also from PR's description).

luisg123v commented on hr_employee_second_lastname/tests/test_hr_employee_lastnames.py:
```diff
@@ -1,10 +1,11 @@
 class TestEmployeeLastnames(TransactionCase):
     def setUp(self):
-        super(TestEmployeeLastnames, self).setUp()
+        super().setUp()
```
Document in commit description

luisg123v commented on hr_employee_second_lastname/hooks.py:
```diff
@@ -0,0 +1,24 @@
+    # (firstname, lastname and lastname2).
+    #
+    # For example:
+    #  After install hr_employee_fisrtname and before install hr_employee_lastnames:
```
Missing renaming module here

luisg123v commented on hr_employee_second_lastname/hooks.py:
```diff
@@ -0,0 +1,24 @@
+    #     firstname = 'John'
+    #     lastname = 'Peterson'
+    #     lastname2 = 'Clinton'
+    env.cr.execute("UPDATE hr_employee SET firstname = NULL, lastname = NULL")
```
Missing renaming module here

luisg123v commented on hr_employee_second_lastname/tests/__init__.py:
```diff
@@ -0,0 +1,3 @@
+# Copyright (C) 2014 Savoir-faire Linux. All Rights Reserved.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from . import test_hr_employee_lastnames
```
And here.

Please fix all similar cases.

andreagidaltig commented:
Hi @luisg123v changes applied, could you review again, please?

luisg123v commented on hr_employee_second_lastname/tests/test_hr_employee_second_lastname.py:
```diff
@@ -0,0 +1,173 @@
+from odoo.tools import submap
+
+
+class TestEmployeeLastnames(TransactionCase):
```
You forgot to update lastnames -> second lastname here.

andreagidaltig replied:
I just updated this, could you review again, please?

luisg123v reviewed (approved):
LGTM :+1: 

@moylop260 could you review/merge, please?

moylop260 commented:
/ocabot merge nobump
